chia/cmds/keys_funcs.py

```diff
@@ -59,29 +58,34 @@ def generate_and_add(label: Optional[str]) -> None:
     """
     unlock_keyring()
     print("Generating private key")
-    query_and_add_private_key_seed(mnemonic=generate_mnemonic(), label=label)
+    query_and_add_key_info(mnemonic_or_pk=generate_mnemonic(), label=label)
 
 
-def query_and_add_private_key_seed(mnemonic: Optional[str], label: Optional[str] = None) -> None:
+def query_and_add_key_info(mnemonic_or_pk: Optional[str], label: Optional[str] = None) -> None:
     unlock_keyring()
-    if mnemonic is None:
-        mnemonic = input("Enter the mnemonic you want to use: ")
+    if mnemonic_or_pk is None:
+        mnemonic_or_pk = input("Enter the mnemonic/observer key you want to use: ")
     if label is None:
         label = input("Enter the label you want to assign to this key (Press Enter to skip): ")
     if len(label) == 0:
         label = None
-    add_private_key_seed(mnemonic, label)
+    add_key_info(mnemonic_or_pk, label)
 
 
-def add_private_key_seed(mnemonic: str, label: Optional[str]) -> None:
+def add_key_info(mnemonic_or_pk: str, label: Optional[str]) -> None:
     """
-    Add a private key seed to the keyring, with the given mnemonic and an optional label.
+    Add a private key seed or public key to the keyring, with the given mnemonic and an optional label.
     """
     unlock_keyring()
     try:
-        sk = Keychain().add_key(mnemonic, label)
-        fingerprint = sk.get_g1().get_fingerprint()
-        print(f"Added private key with public key fingerprint {fingerprint}")
+        if mnemonic_or_pk.count(" ") == 23:
```

Comment:
if someone has an extra `" "` then this could break. Maybe `len(mnemonic_or_pk.strip().split()) == 24`?

Reply:
Fixed